controllers/glance_common.go

```diff
@@ -152,7 +152,7 @@ func GetHeadlessService(
 	helper *helper.Helper,
 	instance *glancev1.GlanceAPI,
 	serviceLabels map[string]string,
-) (ctrl.Result, error) {
+) (ctrl.Result, string, error) {
```

Comment:
Since we are passing instance as a parameter to this function, do we really need to fetch endpointName inside this function and return it (since it is not used for any other purpose), can't we directly use instance.Name outside?

Reply:
I see what you mean, and in theory we can compute it.
However, it's a little price to pay to not have to rewrite (in the controller) the if statement of L161 to distinguish between `split` vs `single` layout.
Since all the logic is already in place here, I can just simplify the logic in the main controller and not have to worry about the nature of the `glanceAPI` (that has an impact over the name: `-external` vs `-internal` vs `-single` vs `-edge`).
Hopefully when we don't have to split (in 19?) we can simplify further this logic!

Reply:
Ack